clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -195,6 +224,43 @@ mlir::Value CIRAttrToValue::visitCirAttr(cir::FPAttr fltAttr) {
       loc, converter->convertType(fltAttr.getType()), fltAttr.getValue());
 }
 
+// ConstArrayAttr visitor
+mlir::Value CIRAttrToValue::visitCirAttr(cir::ConstArrayAttr attr) {
+  auto llvmTy = converter->convertType(attr.getType());
+  auto loc = parentOp->getLoc();
+  mlir::Value result;
+
+  if (auto zeros = attr.getTrailingZerosNum()) {
+    auto arrayTy = attr.getType();
+    result = rewriter.create<mlir::LLVM::ZeroOp>(
+        loc, converter->convertType(arrayTy));
+  } else {
+    result = rewriter.create<mlir::LLVM::UndefOp>(loc, llvmTy);
+  }
+
+  // Iteratively lower each constant element of the array.
+  if (auto arrayAttr = mlir::dyn_cast<mlir::ArrayAttr>(attr.getElts())) {
+    for (auto [idx, elt] : llvm::enumerate(arrayAttr)) {
+      mlir::DataLayout dataLayout(parentOp->getParentOfType<mlir::ModuleOp>());
+      mlir::Value init =
+          emitCirAttrToMemory(parentOp, elt, rewriter, converter, dataLayout);
```

Comment:
I think you can just call `visit(elt)` here. The extra wrapper calls shouldn't be needed.